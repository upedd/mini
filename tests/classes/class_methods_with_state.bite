class Foo {
    bar;
    baz = 10;

    set_bar(bar) {
        this.bar = bar;
    }

    add() {
        return this.bar + this.baz;
    }
}
native print;

let instance = Foo();
print(instance.bar);
print(instance.baz);

instance.set_bar(8);
print(instance.bar);
print(instance.baz);
print(instance.add());

instance.set_bar(-3);
print(instance.bar);
print(instance.baz);
print(instance.add());

let instance2 = Foo();
print(instance2.bar);
print(instance2.baz);

instance2.set_bar(5);
print(instance2.bar);
print(instance2.baz);
print(instance2.add());

print(instance.bar);
print(instance.baz);
print(instance.add());