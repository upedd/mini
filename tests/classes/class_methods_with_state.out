Nil
10
8
10
18
-3
10
7
Nil
10
5
10
15
-3
10
7